rhino/src/main/java/org/mozilla/javascript/config/RhinoProperties.java

```diff
@@ -0,0 +1,163 @@
+ * as service. If no loader was found, the configuration is read from these locations by default:
+ *
+ * <ol>
+ *   <li>rhino.config file from current class' classpath
+ *   <li>rhino.config file from current threas's classpath
+ *   <li>rhino.config file from current directory
+ *   <li>rhino-test.config file from current class' classpath
+ *   <li>rhino-test.config file from current threas's classpath
+ *   <li>rhino-test.config file from current directory
+ *   <li>env variables starting with "RHINO_" (underscores are replaced by '.' and string is
+ *   <li>System-properties starting with "rhino."
+ * </ol>
+ *
+ * <p>(the later config can override previous ones)
+ *
+ * <p>The config files are in UTF-8 format and all keys in this configuration are case-insensitive
+ * and dot/underscore-insensitive.
+ *
+ * <p>
+ *
+ * <p>This means, "rhino.use_java_policy_security" is equvalent to "RHINO_USE_JAVA_POLICY_SECURITY"
+ *
+ * <p>This class contains only the properties. Every plugin etc. can parse its config from there.
+ *
+ * @author Roland Praml, Foconis Analytics GmbH
+ */
+public class RhinoProperties {
+
+    private static String[] CONFIG_FILES = {"rhino.config", "rhino-test.config"};
+    private static final RhinoProperties INSTANCE =
+            AccessController.doPrivileged((PrivilegedAction<RhinoProperties>) () -> init());
+
+    static RhinoProperties init() {
+        RhinoProperties props = new RhinoProperties();
+        Iterator<RhinoPropertiesLoader> loader =
+                ServiceLoader.load(RhinoPropertiesLoader.class).iterator();
+        if (loader.hasNext()) {
+            while (loader.hasNext()) {
+                loader.next().load(props);
+            }
+        } else {
+            props.loadDefaults();
+        }
+        return props;
+    }
+
+    /** Load properties from the default locations. */
+    public void loadDefaults() {
+        ClassLoader classLoader = RhinoProperties.class.getClassLoader();
+        ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
+
+        for (String configFile : CONFIG_FILES) {
+            loadFromClasspath(classLoader, configFile);
+            loadFromClasspath(contextClassLoader, configFile);
+            loadFromFile(new File(configFile));
+        }
+
+        addConfig(System.getenv());
+        addConfig(System.getProperties());
+    }
+
+    private List<Map<?, ?>> configs = new ArrayList<>();
+
+    /** Loads the configuration from the given file. */
+    public void loadFromFile(File config) {
+        if (!config.exists()) {
+            return;
+        }
+        try (InputStream in = new FileInputStream(config)) {
+            Properties props = new Properties();
+            props.load(new InputStreamReader(in, StandardCharsets.UTF_8));
+            addConfig(props);
+        } catch (IOException e) {
+            System.err.println(
```

Comment:
I don't love that we "println" anywhere, but I don't see that we have a choice in this case. But if it happens, it'll be buried in someone's big log file and they won't know what it means -- should we consider adding "Rhino:" or something to identify this message so that people can know why it's appearing?